kordac/processors/utils.py

```diff
@@ -1,49 +1,145 @@
+            raise ArgumentMissingError(processor, argument, "{} is a required argument.".format(argument))
+        elif not is_required and (is_arg or is_flag):
+            dependencies = argument_info.get('dependencies', [])
+            for other_argument in dependencies:
+                if not (parse_argument(other_argument, inputs, None) is not None
+                   or parse_flag(other_argument, inputs) is not None):
+                        message = "{} is a required argument because {} exists.".format(other_argument, argument)
+                        raise ArgumentMissingError(processor, argument, message)
+
+        if is_flag:
+            argument_values[argument] = True
+        elif is_arg:
+            argument_values[argument] = parse_argument(argument, inputs, None)
+    return argument_values
+
+
+def process_parameters(ext, processor, parameters, argument_values):
     '''
-    if not all(parse_argument(parameter, arguments, None) is not None for parameter in required_parameters):
-        parameter = next(parameter for parameter in required_parameters if parse_argument(parameter, arguments, None) is None)
-        raise ParameterMissingError(processor, parameter, "{} is a required parameter.".format(parameter))
+    Processes a given set of arguments by the parameter definitions.
+
+    Args:
+        processor: The processor of the given arguments.
+        parameters: A dictionary of parameter definitions.
+        argument_values: A dictionary of argument to values.
+    Returns:
+        A dictionary of parameter to converted values.
+    '''
+    context = dict()
+    transformations = OrderedDict()
+    for parameter, parameter_info in parameters.items():
+        argument_name = parameter_info['argument']
+        parameter_default = parameter_info.get('default', None)
+        argument_value = argument_values.get(argument_name, parameter_default)
+
+        parameter_value = argument_value
+        if parameter_info.get('transform', None):
+            transform = find_transformation(ext, parameter_info['transform'])
+            if parameter_info.get('transform_condition', None):
+                transformations[parameter] = (eval(parameter_info['transform_condition']), transform)
+            else:
+                transformations[parameter] = (True, transform)
+
+        context[parameter] = parameter_value
+
+    for parameter, (condition, transform) in transformations.items():
+        if context[parameter] is not None:
+            if isinstance(condition, bool) and condition:
+                context[parameter] = transform(context[parameter])
+            if callable(condition) and condition(context):
+                context[parameter] = transform(context[parameter])
+    return context
+
+
+def find_transformation(ext, option):
+    '''
+    Returns a transformation for a given string.
+    TODO:
+    In future should be able to combine piped transformations
+    into a single function.
```

Comment:
The output of one function is the input of another. I imagine you'd write them with a pipe ('|') between them.